docs/plugins/npe2_getting_started.md

```diff
@@ -0,0 +1,286 @@
+the example it will be loaded from:
+`<path/to/my-npy-reader>/my_npy_reader/src/my_npy_reader/napari.yaml`.
+
+The generated `napari.yml` file looks like this:
+
+```yaml
+name: my-npy-reader
+display_name: My Plugin
+contributions:
+  commands:
+    - id: my-npy-reader.get_reader
+      python_name: my_plugin._reader:napari_get_reader
+      title: Open data with napari FooBar
+  readers:
+    - command: my-npy-reader.get_reader
+      accepts_directories: false
+      filename_patterns: ["*.npy"]
+```
+
+With `npe2` installed, we can check that this is a valid plugin manifest:
+
+```bash
+> npe2 validate src/my_npy_reader
+✔ Manifest for 'napari FooBar' valid!
+```
+
+See the {ref}`npe2-manifest-spec` for more details.
+
+## 2. Write code to implement your plugin
+
+For our example `*.npy` reader, edit `src/my_npy_reader/_reader.py`:
+
+```python
+import numpy as np
+
+def npy_file_reader(path):
+    array = np.load(path)
+    # return it as a list of LayerData tuples,
+    # here with no optional metadata
+    return [(array,)]
+
+# This is the function referenced by the reader command in `napari.yml`.
+def napari_get_reader(path):
+    # remember, path can be a list, so we check its type first...
+    # (this example plugin doesn't handle lists)
+    if isinstance(path, str) and path.endswith(".npy"):
+        # If we recognize the format, we return the actual reader function
+        return npy_file_reader
+    # otherwise we return None.
+    return None
+```
+
+You can start testing your plugin using a local editable install, and opening
+napari:
+
+```bash
+pip install -e .
+napari
+```
+
+## 3. Make your plugin discoverable
+
+Packages and modules installed in the same environment as napari may make
+themselves discoverable to napari using package metadata.
+
+The manifest file should be specified in the plugin's `setup.cfg` or
+`setup.py` file using the _[entry point group][epg]_: `napari.manifest`. For
+example, this would be the section for a plugin `npe2-tester` with
+`napari.yaml` as the manifest file:
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    npe2-tester = npe2_tester:napari.yaml
+```
+
+The manifest file is specified relative to the submodule root path.
+So for the example it will be loaded from: `<path/to/npe2-tester>/napari.yaml`.
+
+The manifest file also needs to be included as _[package data][pd]_ in
+distributable forms for the package. For example:
+
+```toml
+[metadata]
+...
+include_package_data=True
+
+[options.package_data]
+npe2_tester =
+    napari.yaml
+```
+
+A user can install napari and your plugin with:
+
+```bash
+pip install napari myplugin
+```
+
+## 4. Preparing for release
+
+Use the `Framework :: napari` [classifier](https://pypi.org/classifiers/) in
+your package's core metadata to make your plugin more discoverable. If you used
+the cookiecutter, this has already been done for you.
+
+Once your package, with its `Framework :: napari` classifier, is listed on
```

Comment:
We need quick instructions here for how to get the package onto pypi e.g. running

```
python setup.py bdist_wheel
twine upload dist/*
```

or using the workflow (can just link to cookiecutter docs here: https://github.com/napari/cookiecutter-napari-plugin/tree/npe2#set-up-automatic-deployments)

Reply:
added a link to the python packaging guide